packages/medusa/src/services/draft-order.ts

```diff
@@ -316,66 +313,65 @@ class DraftOrderService extends BaseService {
         }
       }
 
-      for (const method of shipping_methods) {
-        await this.cartService_
-          .withTransaction(manager)
-          .addShippingMethod(createdCart.id, method.option_id, method.data)
-      }
+      await Promise.all(
+        shipping_methods.map(async (shippingMethod) => {
+          return this.cartService_
+            .withTransaction(transactionManager)
+            .addShippingMethod(createdCart.id, shippingMethod.option_id, shippingMethod.data)
+        })
+      )
 
-      return result
+      return createdDraftOrder
     })
   }
 
   /**
    * Registers a draft order as completed, when an order has been completed.
-   * @param {string} doId - id of draft order to complete
+   * @param {string} draftOrderId - id of draft order to complete
    * @param {string} orderId - id of order completed from draft order cart
-   * @return {Promise} the created order
+   * @return {Promise<DraftOrder>} the created order
    */
-  async registerCartCompletion(doId, orderId) {
-    return this.atomicPhase_(async (manager) => {
-      const draftOrderRepo = manager.getCustomRepository(
-        this.draftOrderRepository_
-      )
-      const draftOrder = await this.retrieve(doId)
-
-      draftOrder.status = "completed"
-      draftOrder.completed_at = new Date()
-      draftOrder.order_id = orderId
-
-      await draftOrderRepo.save(draftOrder)
+  async registerCartCompletion(draftOrderId: string, orderId: string): Promise<DraftOrder> {
+    return this.atomicPhase_(async (transactionManager: EntityManager) => {
+      return transactionManager.update(DraftOrder, {
+        id: draftOrderId
+      }, {
+        status: DraftOrderStatus.COMPLETED,
+        completed_at: new Date(),
+        order_id: orderId
+      })
     })
   }
 
   /**
    * Updates a draft order with the given data
-   * @param {String} doId - id of the draft order
+   * @param {String} id - id of the draft order
    * @param {DraftOrder} data - values to update the order with
    * @return {Promise<DraftOrder>} the updated draft order
    */
-  async update(doId, data) {
-    return this.atomicPhase_(async (manager) => {
-      const doRepo = manager.getCustomRepository(this.draftOrderRepository_)
-      const draftOrder = await this.retrieve(doId)
-      let touched = false
+  async update(id: string, data: DraftOrder): Promise<DraftOrder> {
```

Comment:
@srindom is that normal that this update method does not use the data except the `no_notification_order` in order to update the draftOrder?